test/parallel/test-async-local-storage-exit-does-not-leak.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const { AsyncLocalStorage } = require('async_hooks');
+
+const als = new AsyncLocalStorage();
+
+// The als instance should be getting removed from the storageList in
+// lib/async_hooks.js when exit(...) is called, therefore when the nested runs
+// are called there should be no copy of the als in the storageList to run the
+// _propagate method on.
+als._propagate = common.mustNotCall('_propagate() should not be called');
+
+// We also need to make sure _propagate hasn't been renamed.
+assert.ok(typeof als._propagate === 'function');
```

Comment:
This should be moved up - here it will point to the `mustNotCall` function.

Reply:
Oops, fixed. 😅